org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/ui/text/correction/GetterSetterCorrectionBaseSubProcessor.java

```diff
@@ -0,0 +1,363 @@
+
+	private static class ProposalParameter {
+		public final boolean useSuper;
+		public final ICompilationUnit compilationUnit;
+		public final ASTRewrite astRewrite;
+		public final Expression accessNode;
+		public final Expression qualifier;
+		public final IVariableBinding variableBinding;
+
+		public ProposalParameter(boolean useSuper, ICompilationUnit compilationUnit, ASTRewrite rewrite, Expression accessNode, Expression qualifier, IVariableBinding variableBinding) {
+			this.useSuper= useSuper;
+			this.compilationUnit= compilationUnit;
+			this.astRewrite= rewrite;
+			this.accessNode= accessNode;
+			this.qualifier= qualifier;
+			this.variableBinding= variableBinding;
+		}
+	}
+
+	public static class SelfEncapsulateFieldProposalCore extends ChangeCorrectionProposalCore { // public for tests
+
+		private IField fField;
+
+		public SelfEncapsulateFieldProposalCore(int relevance, IField field) {
+			this(relevance, null, field);
+		}
+
+		public SelfEncapsulateFieldProposalCore(int relevance, Change change, IField field) {
+			super(getDescription(field), change, relevance);
+			fField= field;
+			setCommandId(SELF_ENCAPSULATE_FIELD_COMMAND_ID);
+		}
+
+		public IField getField() {
+			return fField;
+		}
+
+		public static SelfEncapsulateFieldRefactoring getChangeRefactoring(IField field) throws CoreException {
+			final SelfEncapsulateFieldRefactoring refactoring= new SelfEncapsulateFieldRefactoring(field);
+			refactoring.setVisibility(Flags.AccPublic);
+			refactoring.setConsiderVisibility(false);//private field references are just searched in local file
+			refactoring.checkInitialConditions(new NullProgressMonitor());
+			refactoring.checkFinalConditions(new NullProgressMonitor());
+			return refactoring;
+		}
+
+		public static Change getChange(IField field) throws CoreException {
+			Refactoring refactoring= getChangeRefactoring(field);
+			Change createdChange= refactoring.createChange(new NullProgressMonitor());
+			return createdChange;
+		}
+
+		public static TextFileChange getChange(IField field, IFile file) throws CoreException {
+			Change createdChange= getChange(field);
+			if (createdChange instanceof CompositeChange) {
+				for (Change curr : ((CompositeChange) createdChange).getChildren()) {
+					if (curr instanceof TextFileChange && (file == null || ((TextFileChange) curr).getFile().equals(file))) {
+						return (TextFileChange) curr;
+					}
+				}
+			}
+			return null;
+		}
+
+
+		public static String getDescription(IField field) {
+			return Messages.format(CorrectionMessages.GetterSetterCorrectionSubProcessor_creategetterunsingencapsulatefield_description, BasicElementLabels.getJavaElementName(field.getElementName()));
+		}
+
+		/*
+		 * @see org.eclipse.jface.text.contentassist.ICompletionProposalExtension5#getAdditionalProposalInfo(org.eclipse.core.runtime.IProgressMonitor)
+		 * @since 3.5
```

Comment:
Please remove the whole comment here:
* Current version of the bundle is 1.21 thus the since is totally wrong. 
* The see tag is useless either as ICompletionProposalExtension5 is not visible in this bundle.